tests/api_tests/test_biobank_specimen_api.py

```diff
@@ -0,0 +1,162 @@
+            for status_field in ['status', 'freezeThawCount', 'location', 'quantity', 'quantityUnits',
+                                  'processingCompleteDate', 'deviations']:
+                if status_field in test_json:
+                    self.assertEqual(test_json['status'][status_field], specimen_json['status'][status_field])
+
+    def retrieve_specimen_json(self, specimen_id, order_id):
+        specimen = self.dao.get((specimen_id, order_id))
+        json = self.dao.to_client_json(specimen)
+        return json
+
+    def test_put_new_specimen_minimal_data(self):
+        payload = {
+            'rlimsID': 'sabrina',
+            'orderID': self.bio_order.biobankOrderId,
+            'participantID': config_utils.to_client_biobank_id(self.participant.biobankId),
+            'testcode': 'test 1234567'
+        }
+        rlims_id = payload['rlimsID']
+        result = self.send_put(f"Biobank/specimens/{rlims_id}", request_data=payload)
+
+        saved_specimen_client_json = self.retrieve_specimen_json(result['id'], result['orderID'])
+        self.assertSpecimenJsonMatches(saved_specimen_client_json, payload)
+
+    def test_put_new_specimen_all_data(self):
+        payload = {
+            'rlimsID': 'sabrina',
+            'orderID': self.bio_order.biobankOrderId,
+            'participantID': config_utils.to_client_biobank_id(self.participant.biobankId),
+            'testcode': 'test 1234567',
+            'repositoryID': 'repo id',
+            'studyID': 'study id',
+            'cohortID': 'cohort id',
+            'sampleType': 'sample',
+            'status': {
+                'status': 'good',
+                'freezeThawCount': 1,
+                'location': 'Greendale',
+                'quantity': '1',
+                'quantityUnits': 'some units',
+                'processingCompleteDate': TIME_2.isoformat(),
+                'deviations': 'no deviation'
+            },
+            'collectionDate': TIME_1.isoformat(),
+            'confirmationDate': TIME_2.isoformat()
+        }
+        rlims_id = payload['rlimsID']
+        result = self.send_put(f"Biobank/specimens/{rlims_id}", request_data=payload)
+
+        saved_specimen_client_json = self.retrieve_specimen_json(result['id'], result['orderID'])
+        self.assertSpecimenJsonMatches(saved_specimen_client_json, payload)
+
+    def test_put_specimen_exists(self):
+        payload = {
+            'rlimsID': 'sabrina',
+            'orderID': self.bio_order.biobankOrderId,
+            'participantID': config_utils.to_client_biobank_id(self.participant.biobankId),
+            'testcode': 'test 1234567'
+        }
+        rlims_id = payload['rlimsID']
+        initial_result = self.send_put(f"Biobank/specimens/{rlims_id}", request_data=payload)
+
+        new_payload = payload
+        new_payload['testcode'] = 'updated testcode'
+        self.send_put(f"Biobank/specimens/{rlims_id}", request_data=new_payload, headers={"if-match": 'W/"1"'})
```

Comment:
We don't need "if-match" header for this API.

Reply:
Ah, sorry I missed this. Fixing now.